### PR TITLE
[pkg/telemetryquerylanguage] Add TQL mapping functions for InstrumentationScope attributes

### LINKED ISSUE #13639: [transformprocessor/TQL] Instrumentation scope attributes are not accessible
**Describe the bug**
When using TQL in the transformprocessor, `instrumentation_scope.attributes` and `instrumentation_scope.attributes[""]` are not available, since the respective mapping functions are not implemented in the TQL contexts (https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/telemetryquerylanguage/contexts) 

**What did you expect to see?**
Using `instrumentation_scope.attributes` should give access to scope attributes.

**What did you see instead?**
Collector did not start: 
```
otel-collector_1  | Error: failed to get config: invalid configuration: processor "transform" has invalid configuration: invalid argument at position 0 invalid path expression [{instrumentation_scope <nil>} {attributes <nil>}]
otel-collector_1  | 2022/08/26 15:37:39 collector server run finished with error: failed to get config: invalid configuration: processor "transform" has invalid configuration: invalid argument at position 0 invalid path expression [{instrumentation_scope <nil>} {attributes <nil>}]
```

**What version did you use?**
`otel/opentelemetry-collector-contrib:0.58.0`

**What config did you use?**
Config: 
```yaml
processors:
  transform:
    metrics:
      queries:
        - keep_keys(instrumentation_scope.attributes, "scope-attribute-1")
```

**Environment**
Docker: `otel/opentelemetry-collector-contrib:0.58.0`

### PR DESCRIPTION
**Pull request by pirgeo**
**Description:**
Adds mapping functions for InstrumentationLibraryScope.

**Link to tracking Issue:** 
Resolves #13639 

**Documentation:** Updated the respective README.md files.

### CODE REVIEW

**Review by TylerHelmuth — APPROVED** [review]
Thanks for keeping the TQL up-to-date with the Spec!

**Comment by TylerHelmuth** [comment]
@pirgeo please add a changelog entry for this change

File: unreleased/instrumentation-scope-attributes.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: transformprocessor
```
Comment (TylerHelmuth) [inline unreleased/instrumentation-scope-attributes.yaml:5]:
```suggestion
component: pkg/telemetryquerylanguage
```

**Review by TylerHelmuth — APPROVED** [review]

**Review by bogdandrutu — DISMISSED** [review]
Blocking this because we need to stop adding copy paste code. Waiting for https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/13650

**Comment by bogdandrutu** [comment]
@pirgeo please rebase this.

**Comment by pirgeo** [comment]
Since the underlying change made this a very small change, I reorganized the commits a bit.

**Review by TylerHelmuth — COMMENTED** [review]
Can you add new tests scenarios in scope_test.go for the new getters/setters?

**Review by TylerHelmuth — APPROVED** [review]

**Comment by TylerHelmuth** [comment]
@pirgeo can you address the build failures?

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**